shared/teams/teamname.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+// This logic is copied from go/protocol/keybase1/extras.go.
+
+const validTeamnamePart = (s: string): boolean => {
+  if (s.length < 2 || s.length > 16) {
+    return false
+  }
+
+  return /^([a-zA-Z0-9][a-zA-Z0-9_]?)+$/.test(s)
+}
+
+const validTeamname = (s: string): boolean => {
+  return s.split('.').every(validTeamnamePart)
+}
+
+type Teamname = string
```

Comment:
If I instead import `Teamname` from `../constants/teams`, running `yarn test` gets me:

```
  ● Test suite failed to run

    TypeError: Teams.makeState is not a function

      at Object.<anonymous> (constants/entities.js:104:13)
      at Object.<anonymous> (constants/reducer.js:7:15)
      at Object.<anonymous> (constants/selectors.js:4:1)
      at Object.<anonymous> (constants/search.js:3:16)
      at Object.<anonymous> (constants/chat.js:86:239)
      at Object.<anonymous> (constants/teams.js:3:11)
      at Object.<anonymous> (teams/teamname.js:3:1)
      at Object.<anonymous> (teams/__test__/teamname.test.js:3:15)
          at Generator.next (<anonymous>)
          at new Promise (<anonymous>)
          at Generator.next (<anonymous>)
          at <anonymous>
```

which is weird.

Reply:
hmm, maybe related to https://github.com/keybase/client/pull/9379?

Reply:
From discussion in chat, it sounds like this stems from some import cycles. Maybe this'll go away once @chrisnojima lands his changes removing some cycles...

Reply:
Sadly, rebasing didn't help.